src/routes/config.ts

```diff
@@ -21,13 +20,13 @@ export const vars: RequestHandler = (req, res) => {
 	const schema: JSONSchema6 = {
 		type: 'object',
 		$schema: 'http://json-schema.org/draft-06/schema#',
-		properties: {
-			...SUPERVISOR_CONFIG_VAR_PROPERTIES,
-
-			...(RESIN_HOST_CONFIG_CAPABLE_DEVICE_TYPES.includes(req.query.deviceType)
-				? HOST_CONFIG_VAR_PROPERTIES
-				: {}),
-		},
+		properties: Object.assign(
+			{},
+			SUPERVISOR_CONFIG_VAR_PROPERTIES,
+			...DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES.filter(config =>
+				config.capableDeviceTypes.has(req.query.deviceType),
+			).map(config => config.properties),
+		),
 	};
```

Comment:
```suggestion
		properties: {
			...SUPERVISOR_CONFIG_VAR_PROPERTIES,
			...DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES.filter(config =>
				config.capableDeviceTypes.has(req.query.deviceType),
			).map(config => config.properties),
		},
	};
```

Reply:
That doesn't work @Page- , since `SUPERVISOR_CONFIG_VAR_PROPERTIES` is an object, while `DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES.filter(...).map(...)` is an array of objects and that array get destructured as "rest arguments" of the `Object.assign`.
I don't think that we can do `{...x}` destructuring of unknown number of objects w/o a loop, which would be bad, since it would create tons of intermediate short lived objects.